### PR TITLE
KFSPTS-21564: Modify code to use EGAs instead of netids for From email addresses when email sent from KFS. Required for Cornell AppSMTP service change.

### PR DESCRIPTION
**Pull request by nkimble**
Please review this code but DO NOT merge yet.
Item 1, Item 4, and Item 6 as discussed in Description for sub-task KFSPTS-21577 have been addressed by these code changes.

### CODE REVIEW

File: src/main/java/edu/cornell/kfs/krad/service/impl/CuKualiExceptionIncidentServiceImpl.java
Comment (cah292) [inline src/main/java/edu/cornell/kfs/krad/service/impl/CuKualiExceptionIncidentServiceImpl.java:11]: Some of these comments should be cleaned up. The original code in commented form can be removed, since we can go to the superclass to view it if needed. It's also not clear about how "kr.incident.mailing.list" relates to this unless you look at the bean file.

In addition, you need to override the appropriate service bean so that it will use this new subclass. (I think "cf.kradExceptionIncidentService" is the bean name.)

File: src/main/java/edu/cornell/kfs/krad/service/impl/CuKualiExceptionIncidentServiceImpl.java
Comment (cah292) [inline src/main/java/edu/cornell/kfs/krad/service/impl/CuKualiExceptionIncidentServiceImpl.java:7]: Minor typo in comment.

**Review by cah292 — CHANGES_REQUESTED** [review]
I found a few things that should be updated, but otherwise the changes look good.

File: src/main/java/edu/cornell/kfs/krad/service/impl/CuKualiExceptionIncidentServiceImpl.java
Comment (nkimble) [reply src/main/java/edu/cornell/kfs/krad/service/impl/CuKualiExceptionIncidentServiceImpl.java]: Comments clean up. Thanks for catching the bean definition.

File: src/main/resources/edu/cornell/kfs/krad/config/CUKRADSpringBeans.xml
Comment (cah292) [inline src/main/resources/edu/cornell/kfs/krad/config/CUKRADSpringBeans.xml:76]: It looks like the base financials version of the "cf.kradExceptionIncidentService" bean does not follow the "parentBean" convention. You may have to include the full XML of the base "cf.kradExceptionIncidentService" bean when you customize it here.

**Review by cah292 — CHANGES_REQUESTED** [review]
I found one more item that needs fixing, but otherwise the changes look good.

File: src/main/resources/edu/cornell/kfs/krad/config/CUKRADSpringBeans.xml
Comment (nkimble) [reply src/main/resources/edu/cornell/kfs/krad/config/CUKRADSpringBeans.xml]: fixed

**Review by cah292 — CHANGES_REQUESTED** [review]
The service bean looks good, but the unit tests are failing because the bean's file doesn't have a namespace entry for the "p" prefix. Please update "CUKRADSpringBeans.xml" to include the appropriate "xmlns:p" entry at the top of the file.

**Review by cah292 — COMMENTED** [review]
Changes look good, and unit tests pass. I will wait for confirmation that we're ready to move forward with the fixes before merging.

**Review by cah292 — APPROVED** [review]
I've received confirmation from Nancy that we are ready to move these changes forward. Merging.

**Pull request merged**
